src/components/MFCard/DynamicCardIframe.tsx

```diff
@@ -25,11 +25,36 @@ const DynamicCardIframe: React.FC<Props> = ({ task, hash }) => {
       };
 
       fetch(`${apiHttp(taskCardsPath(task))}?invalidate=true`)
-        .then((result) => result.json())
         .then((result) => {
+          if (!result.ok) {
+            throw new Error('Failed to fetch');
+          }
+          return result.json();
+        })
+        .then((result) => {
+          // check if the data attribute exists and is not null
+          if (result.data === null) {
+            timeoutRef.current = setTimeout(() => {
+              getNewData(token, updateFn, iframe);
+            }, DYNAMIC_CARDS_REFRESH_INTERVAL);
+            return;
+          }
           if (token === result.data?.reload_token) {
-            // Call the callback provided by the card to add new data
-            updateFn(result.data?.data);
+            let runUpdate = false;
+            if (result.data?.created_on) {
```

Comment:
This logic is a bit tricky to understand. Simplify it, or add some comments. You can easily get rid of the `runUpdate` variable and just call `updateFn(result.data?.data)` where needed